torch/csrc/jit/codegen/cuda/scheduler/registry.h

```diff
@@ -8,6 +8,90 @@ namespace jit {
+//!    segmenter and schedulers.
+//!  It is important that input id encoding should be up to date with any change
+//!   of this class to avoid launching compiled kernels with illegal inputs.
+class TORCH_CUDA_CU_API SchedulerRuntimeInfo {
+ public:
+  // This constant defines the widest vectorized load/store we
+  //  can do on the target. We should be getting this from some
+  //  library calls. We want to use max vectorized load when we can while
+  //  we don't want to make the max conservatively large since it may create too
+  //  many cache entries for different alignments.
+  const size_t max_alignment_size_in_byte = 128;
+
+  //! Create runtime info for given fusion and input. Creating and binding
+  //! evaluator is
+  //!  optional. The evaluator is used to manage intermediate integers in the
+  //!  fusion. We need them for segmenter and schedulers, but we don't need them
+  //!  when we are just using this class to provide additional encoding for
+  //!  kernel cache lookup.
+  SchedulerRuntimeInfo(
+      Fusion* complete_fusion,
+      const at::ArrayRef<IValue>& inputs,
+      bool create_expr_evaluator = false);
+
+  //! Create runtime info by copying all the global input meta data (i.e.
+  //! alignment), but not
+  //!  the expression evaluator.
+  SchedulerRuntimeInfo(const SchedulerRuntimeInfo& global_runtime_info);
+
+  //! Segmenter facing interface for copying integer values on the segmented
+  //! group inputs
+  //!  so that the intermediates can be computed within the boundary of the
+  //!  segmented group
+  void bindGroupInputs(
+      SegmentedGroup* group,
+      SchedulerRuntimeInfo& global_runtime_info);
+
+  //! Lookup for the alignment sizes of the given tv. Currently only different
+  //! for full fusion
+  //!  tensors and returns a default value for others
+  size_t getAlignmentSize(TensorView* tv);
+
+  //! Take the minimum of input tv alignment sizes. This is both information for
+  //! vectorization and
+  //!  a signature for kernel cache id lookup. May need to be updated with
+  //!  vectorization logic.
+  size_t getCommonAlignmentSize() const {
+    return common_alignment_size_;
+  }
+
+  Fusion* fusion() {
+    return complete_fusion_;
+  }
+
+  ExpressionEvaluator& expressionEvaluator() {
+    TORCH_INTERNAL_ASSERT(expression_evaluator_ != nullptr);
+    return *expression_evaluator_;
+  }
+
+ private:
+  // Bind full fusion inputs to the internal expression evaluator
+  void initializeExpressionEvaluator(const at::ArrayRef<IValue>& inputs);
+
+  // Compute alignment data for all input tensors of full fusion
+  void collectAlignmentInfo(const at::ArrayRef<IValue>& inputs);
+
+  // Compute alignment data for given tensor
+  size_t collectAlignmentSize(const at::Tensor& tensor) const;
+
+ private:
+  std::unique_ptr<ExpressionEvaluator> expression_evaluator_ = nullptr;
+  Fusion* complete_fusion_;
+  std::unordered_map<TensorView*, size_t> alignment_map_;
```

Comment:
What about contiguity information on the inputs?

Reply:
Maybe @jjsjann123 has an idea if we can just make a copy of the metadata without taking ownership of the tensor. Then simply have a map from TensorView* to the meta data.

Reply:
Good catch. Just added strides to `SchedulerRuntimeInfo` for the input tensors. Please let me know if more data is needed here for vectorization.